src/collection.rs

```diff
@@ -442,14 +452,110 @@ impl<'a, C: ClientExt> Collection<'a, C> {
     /// Creates a new document from the document given in the body, unless
     /// there is already a document with the _key given. If no _key is given, a
     /// new unique _key is generated automatically.
+    /// Possibly given _id and _rev attributes in the body are always ignored,
+    /// the URL part or the query parameter collection respectively counts.
+    ///
+    /// If the document was created successfully, then the Location header contains
+    /// the path to the newly created document.
+    /// The Etag header field contains the revision of the document.
+    /// Both are only set in the single document case.
+    ///
+    /// If silent is not set to true, the body of the response contains a JSON object with the following attributes:
     ///
+    /// _id contains the document identifier of the newly created document
+    /// _key contains the document key
+    /// _rev contains the document revision
+    /// If the collection parameter waitForSync is false, then the call returns as soon as the document has been accepted.
+    /// It will not wait until the documents have been synced to disk.
+    ///
+    /// Optionally, the query parameter waitForSync can be used to force synchronization of the document creation
+    /// operation to disk even in case that the waitForSync flag had been disabled for the entire collection.
+    /// Thus, the waitForSync query parameter can be used to force synchronization of just this specific operations.
+    /// To use this, set the waitForSync parameter to true. If the waitForSync parameter is not specified or set to false,
+    /// then the collection’s default waitForSync behavior is applied.
+    /// The waitForSync query parameter cannot be used to disable synchronization for collections that have a default waitForSync value of true.
+    ///
+    /// If the query parameter returnNew is true, then, for each generated document, the complete new document is returned under the new attribute in the result.
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn create_document<T>(&self, _doc: Document<T>) {
-        unimplemented!()
+    pub async fn create_document<T>(
+        &self,
+        _doc: Document<T>,
```

Comment:
1. a little style preference: use `doc` instead of `_doc` please. The original `_doc` name is to stop cargo fmt from complaining unused parameter.

2. a little confusion: we don't know `_id`, '_key', '_rev' before creating the document, and I don't think most users want to assign these field manually. Maybe use `doc: T` instead? Also, we don't need the `DeserializeOwned` trait here since users might want to use different type for serialization and deserialization.

Reply:
Hello !

Okay , i ll do the changes !
Regarding the deserializedOwned, it was because the compiler asked for it , I ll' see if I can remove it.

By the way, I do not succeed to implement `read document header`

I got some 

```
Err(Serde(Error("EOF while parsing a value", line: 1, column: 0)))
thread 'test_get_read_document_header' panicked at 'called `Result::unwrap()` on an `Err` value: Serde(Error("EOF while parsing a v
```

Reply:
Trait bould should restricted to minimum, and I don't think `Debug` trait is a must either.

Reply:
Maybe just deserialize the `_id`, `_rev` and `_key` fileds? And simply construct a document by ourselves?

```rust
    pub async fn create_document<T>(
        &self,
        doc: Document<T>,...) -> Result<Document<T>, ClientError>
where T: Serialize{
// ... send request and get response
        let header: DocumentHeader =
            serde_json::from_str(self.session.post(url, body.as_str()).await?.text())?;
Ok( Document{ document: doc, _rev: ..., _id:..., _key:...)

}
```

Reply:
Yes true. Debug is not necessary :)

By the way, I have just edited my answer from before :)

Reply:
> I got some error when trying to use Option
> 
> ```
> Err(Serde(Error("EOF while parsing a value", line: 1, column: 0)))
> thread 'test_get_read_document_header' panicked at 'called `Result::unwrap()` on an `Err` value: Serde(Error("EOF while parsing a v
> ```

Remember to add `#[serde(default)]` to `Option` type field.

Reply:
> By the way, I do not succeed to implement `read document header`

That's is required by `Document`, and it's reasonable for Document to require both `Serialize` and `Deserialize`. But here I think we can simply pass `T` instead of `Docuemnt<T>` to `create_document`. If a user want to manually assign `_id`, '_rev' and '_key', he/she can pass a `Document<DataType>`, which certainlly meet our requirement of `T` becasue a `Document` implemented `Serialize`.

Reply:
> 2- In the Arangojs driver, users have the possibility to choose the metadata with `DocumentData<T>`
> It happened for me in the past to need to use the _key when creating a doc. It is also written in the doc and used in similar libraries.
> I would advise that we stay with the doc: `Document<T>` so we stay consistent with similar libraries regarding expectations and use.

I think simply pass `T` instead of `Docuemnt<T>` to `create_document` suffices.
 
If a user want to manually assign `_id`, '_rev' and '_key', he/she can pass a `Document<DataType>`, which certainlly meet our requirement of `T` becasue a `Document` implemented `Serialize`. In other words, `Document<DataType>` is compatible with `T`. So the case that need to use the _key when creating a doc CAN be archived. And we just need several lines of explanation on document I think.

Reply:
Okay :)
Thank you a lot for the feedback !